tests/model/test_pytorch_model.py

```diff
@@ -72,20 +72,25 @@ def new_local_optimizer(*args, **kwargs):
             return mock_local_optimizer
 
         pytorch_model_setup.model.new_local_optimizer = new_local_optimizer
-        bridges.sgd_bridge().do_sgd(
-            pytorch_model_setup.model, user_dataset,
+        # This is same as bridges.sgd_bridge().do_sgd, but we want
+        # to check the returned metadata as well.
+        from pfl.internal.bridge.pytorch.sgd import _sgd_train_step
+        train_metadata = pytorch_model_setup.model.do_multiple_epochs_of(
+            user_dataset,
             NNTrainHyperParams(
                 local_learning_rate=local_learning_rate,
                 local_num_epochs=local_num_epochs,
                 local_batch_size=1,
-                grad_accumulation_steps=grad_accumulation_steps))
+                grad_accumulation_steps=grad_accumulation_steps),
+            _sgd_train_step)
 
         # Check if optimizer step is called correct number of times
         total_steps = 2 * local_num_epochs
```

Comment:
I was confused about the "2" here. It's the `len(user_dataset) / local_batch size`, correct? I think writing it like this would make it much more legible in the tests.